nup/marshalJSON.go

```diff
@@ -69,10 +77,10 @@ func MarshalJSON(v interface{}) ([]byte, error) {
 			return nil
 		}
 		switch field := fieldValue.Addr().Interface().(type) {
-		case Nullable:
-			// Only marshal nullable fields that are explicitly set.
-			if field.IsSet() {
-				if err := appendField(field.InterfaceValue()); err != nil {
+		case updateMarshaller:
```

Comment:
Users no longer need to implement new update types, so we don't need `Nullable` anymore. However, we still need an unexported interface for marshalling purposes, just to checking if an update field is a set/remove operation and getting its value as an `interface{}`.

I would like to just match `Update` and `SliceUpdate` values and skip this rather meaningless interface, but as far as I can tell there's currently no way to match against generic types in a type switch statement. You can match against an _instance_ of a generic type, e.g. `Update[int]`, but not against the generic type itself, e.g. `Update`/`Update[T any]`. Maybe a future go version will add support for that kind of thing.